### ISSUE TITLE
Pursuit-evasion observation and action spaces

### ISSUE THREAD

**Issue by bbrighttaer** [sisl/MADRL#15]
Hello, I have been spending some time now to understand the meaning behind the pursuit-evasion observation matrix but I don't seem to get a clear picture of it. For instance, for a 2 pursuers vs 5 evaders game, I get an observation like
`observation: [array([[[0. , 0. , 0. , 0. ],
        [0. , 0. , 0. , 0. ],
        [0. , 0. , 0. , 0. ]],

       [[0. , 0. , 0. , 0. ],
        [0. , 0.1, 0. , 0. ],
        [0. , 0. , 0. , 0. ]],

       [[0. , 0. , 0.1, 0. ],
        [0. , 0. , 0. , 0. ],
        [0. , 0. , 0. , 0. ]]]), array([[[0.1, 0. , 0. , 0. ],
        [0. , 0. , 0. , 0. ],
        [0. , 0. , 0. , 0. ]],

       [[0.1, 0. , 0. , 0. ],
        [0. , 0.1, 0. , 0.5],
        [0. , 0. , 0. , 0. ]],

       [[0.1, 0. , 0. , 0. ],
        [0. , 0. , 0. , 0. ],
        [0. , 0. , 0. , 0. ]]])]`
in one iteration. The environment was creating using:
`env = PursuitEvade([map_mat], n_evaders=n_evaders, n_pursuers=n_pursuers, obs_range=3, n_catch=2, surround=False,
                       reward_mech='local')`
and I have set `self.flatten = False` to help me view the individual channels. The pursuers are select random actions in this instance.

 I wanna know what each channel represents in this case and the meaning of the matrix elements. I have read the paper and it states that each agent recieves a limited-range observation consisting of location of pursuers, evaders, and obstacles. However, I do not know in which order they appear in the experiment and I'm unable to deduce the meaning of the channels from the rendered environment.
Also, I have realized that the 5 possible actions are assigned nominal values (0-4). Please, which value corresponds to which action in this case?
Thanks.

**Comment by rejuvyesh** [comment]
Checkout https://github.com/sisl/MADRL/blob/a3651e225c388c8eb42142a4764fb3f080f07def/madrl_environments/pursuit/pursuit_evade.py#L438-L441

where model_state is https://github.com/sisl/MADRL/blob/a3651e225c388c8eb42142a4764fb3f080f07def/madrl_environments/pursuit/pursuit_evade.py#L244-246

 ```
actions =         [0, # move left
                         1, # move right
                         2, # move up
                         3, # move down
                         4] # stay
 ```

**Comment by bbrighttaer** [comment]
Thanks very much. I also found avoiding the rollaxis [https://github.com/sisl/MADRL/blob/a3651e225c388c8eb42142a4764fb3f080f07def/madrl_environments/pursuit/pursuit_evade.py#L449](url) and maintaining the 4 channels very insightful. 

However, I would want to know if the following observation of mine is right and if so why the channels are transformed.
I realized this work is linked to an earlier study http://cs231n.stanford.edu/reports/2016/pdfs/122_Report.pdf and section 3.2 of that study explains the 4 channels clearly and if that's the case, I think rolling the axis rather mixes the information of the channels. 
For instance, if A is initialized to be of size (4,3,3):
`A = array([[[3, 1, 2],
        [1, 1, 2],
        [3, 2, 1]],

       [[3, 0, 3],
        [2, 0, 2],
        [0, 1, 0]],

       [[4, 0, 0],
        [4, 4, 0],
        [1, 4, 0]],

       [[0, 2, 2],
        [2, 4, 4],
        [4, 0, 1]]])`
assuming each channel presents a different kind of information,
`np.rollaxis(A,0,3)` yields:
array([[[3, 3, 4, 0],
        [1, 0, 0, 2],
        [2, 3, 0, 2]],

       [[1, 2, 4, 2],
        [1, 0, 4, 4],
        [2, 2, 0, 4]],

       [[3, 0, 1, 4],
        [2, 1, 4, 0],
        [1, 0, 0, 1]]])
which I think spreads the different kinds of information across all channels.
